### PR TITLE
docs: adding general rules to e2e contribution guidelines 

### PR DESCRIPTION
**Pull request by nilgaar**
## Description

Adding some general rules that I think are relevant.

## Which issue(s) does this PR fix

- Fixes https://issues.redhat.com/browse/RHIDP-4243

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: docs/e2e-tests/CONTRIBUTING.MD [hunk 1]
```diff
@@ -37,6 +37,13 @@ These principles are valid for new contributions. Some parts of the codebase may
 
    All architecture principles and guidelines must be agreed upon by the team. Every team member's voice is important, and open communication ensures alignment and shared understanding of the project's direction.
 
+6. **General Rules**
+
+   All tests have to follow a set of simple rules:
+   I. Check the preconditions. If tests are running in parallel, try to create the preconditions you need to avoid race conditions.
+   II. Each test has to test one and only one thing.
+   III. All tests have to contain preconditions, actions, and assertions. _Think in [Gherkin](https://cucumber.io/docs/guides/overview/) terms_.
```
Comment (subhashkhileri) [inline docs/e2e-tests/CONTRIBUTING.MD:45]: @nilgaar isn't Gherkin specific to cucumber framework?

Comment (nilgaar) [reply docs/e2e-tests/CONTRIBUTING.MD:45]: Yes! I totally is.
My point here is that Gherkin is a great way to structure and think about a test: Checking that you have what you need, performing an action and doing an assertion.
Maybe the phrase can be improved a little ,

**Review by josephca — APPROVED** [review]
/lgtm

**Pull request merged**